### PR TITLE
fix: the judgement of the resource drifted

### PR DESCRIPTION
**Pull request by aFlyBird0**
Signed-off-by: Bird <aflybird0@gmail.com>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description
See #975 for details.

### CODE REVIEW

**Review by daniel-hutao — APPROVED** [review]
maybe the file `test_drifted_state` could have a better name.

**Comment by aFlyBird0** [comment]
> maybe the file `test_drifted_state` could have a better name. 

Thank you, done.

File: internal/pkg/pluginengine/change_helper.go [hunk 1]
```diff
@@ -42,6 +43,29 @@ func generateAction(tool *configmanager.Tool, action statemanager.ComponentActio
...
+	if err != nil {
+		return true, err
+	}
+	fromReadWithoutType := map[string]interface{}{}
+	err = yaml.Unmarshal(fromReadBytes, &fromReadWithoutType)
+	if err != nil {
+		return true, err
+	}
+	log.Info(fromStateFile)
+	log.Info(fromReadWithoutType)
```
Comment (iyear) [inline internal/pkg/pluginengine/change_helper.go:63]: Will these two printing with info level cause too much output?

**Review by iyear — APPROVED** [review]
Well Done!

File: internal/pkg/pluginengine/change_helper.go (hunk 1 above)
Comment (aFlyBird0) [reply internal/pkg/pluginengine/change_helper.go:63]: > Will these two printing with info level cause too much output?

Sorry, it is for the test. I forget to delete them. You are very careful!

**Pull request merged**